ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -342,86 +378,120 @@ class TreegraphSeries extends OrganizationSeries {
-            to: nodeColumns.length - 1, // Height of the tree
+            to: tree.height,
             defaults: {
-                borderColor: options.borderColor,
-                borderRadius: options.borderRadius, // organization series
-                borderWidth: options.borderWidth,
-                color: series.color,
-                colorByPoint: options.colorByPoint,
-                // NOTE: if support for allowTraversingTree is added, then
-                // levelIsConstant should be optional.
-                levelIsConstant: true,
-                linkColor: options.linkColor, // organization series
-                linkLineWidth: options.linkLineWidth, // organization series
-                linkOpacity: options.linkOpacity,
-                states: options.states
+                levelIsConstant: series.options.levelIsConstant,
+                colorByPoint: options.colorByPoint
             }
-        });
+        }) as any;
+
+        series.setTreeValues(tree);
 
         this.layoutAlgorythm.calculatePositions(series);
-        series.layoutModifier = this.getLayoutModifiers();
+        series.layoutModifier = this.getLayoutModifier();
 
-        // First translate all nodes so we can use them when drawing links.
-        this.nodes.forEach((node): void => {
-            series.translateNode(node);
+        this.points.forEach((point): void => {
+            this.translateNode(point);
         });
 
-        // Then translate links.
-        this.points.forEach(function (linkPoint): void {
-            // If weight is 0 - don't render the link path #12453,
-            // render null points.
-            if ((linkPoint.weight || linkPoint.isNull) && linkPoint.to) {
-                linkPoint.formatPrefix = 'link';
-                series.translateLink(linkPoint);
-                linkPoint.allowShadow = false;
-            }
+        this.links.forEach((link): void => {
+            this.translateLink(link);
         });
     }
 
+    public translateLink(link: TreegraphLink): void {
+        OrganizationSeries.prototype.translateLink.apply(this, arguments);
+        if (link.dlBox) {
+            link.dlBox.centerX = link.dlBox.x;
+        }
+    }
+
     /**
      * Treegraph has two separate collecions of nodes and lines,
      * render dataLabels for both sets.
      */
     public drawDataLabels(): void {
         if (this.options.dataLabels) {
-            const textPath = this.options.dataLabels.textPath;
+            const textPath = (this as any).options.dataLabels.textPath;
 
             // Render node labels.
             super.drawDataLabels.apply(this, arguments);
 
             // Render link labels.
-            this.points = this.data;
-            this.options.dataLabels.textPath =
-                this.options.dataLabels.linkTextPath;
-            super.drawDataLabels.apply(this, arguments);
+            const points = this.points;
+            this.points = this.links as any;
```

Comment:
I can imagine why we need `as any` in here but maybe lets at least cast to proper points type?